code/modules/sanity/sanity_mob.dm

```diff
@@ -237,12 +255,22 @@
 /datum/sanity/proc/onSeeDeath(mob/M)
 	if(ishuman(M))
 		var/penalty = -SANITY_DAMAGE_DEATH(owner.stats.getStat(STAT_VIG))
+		if(special_death_view_effect)
```

Comment:
Why is it even needed, if Nihilist is the only perk that can provide such effect? I also don't see where this perk changes this var. You better just check for perk itself here.